packages/dicomImageLoader/src/imageLoader/wadors/loadImage.ts

```diff
@@ -90,68 +119,126 @@ function loadImage(
 
   const start = new Date().getTime();
 
-  const promise = new Promise<DICOMLoaderIImage>((resolve, reject) => {
-    // TODO: load bulk data items that we might need
-
-    // Uncomment this on to test jpegls codec in OHIF
-    // const mediaType = 'multipart/related; type="image/x-jls"';
-    // const mediaType = 'multipart/related; type="application/octet-stream"; transfer-syntax="image/x-jls"';
-    const mediaType =
-      'multipart/related; type=application/octet-stream; transfer-syntax=*';
-    // const mediaType =
-    //   'multipart/related; type="image/jpeg"; transfer-syntax=1.2.840.10008.1.2.4.50';
-
-    function sendXHR(imageURI: string, imageId: string, mediaType: string) {
+  const { retrieveType, transferSyntaxUID } = options;
+  const loaderOptions = getOptions();
+  options.retrieveOptions = loaderOptions.getRetrieveOptions(
+    retrieveType,
+    transferSyntaxUID
+  );
+  const uncompressedIterator = new ProgressiveIterator<DICOMLoaderIImage>(
+    'decompress'
+  );
+  async function sendXHR(imageURI: string, imageId: string, mediaType: string) {
```

Comment:
Similarly, like viewports, we should have separate branches for non-progressive and progressive approaches. Ideally, we should avoid mixing them here to enhance the debugging experience for regular use cases.